daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/NumericOps.scala

```diff
@@ -174,13 +174,7 @@ case object TimesLong extends NumericOp {
   def operate(v1: JNumber, v2: JNumber): JNumber = { asLong(v1) * asLong(v2) }
 }
 case object DivLong extends NumericOp {
-  def operate(v1: JNumber, v2: JNumber): JNumber = {
-    val v2l = asLong(v2)
-    if (v2l == 0) {
-      throw new ArithmeticException("/ by zero")
-    }
-    asDouble(v1) / v2l
-  }
+  def operate(v1: JNumber, v2: JNumber): JBigDecimal = { DivDecimal.operate(v1, v2) }
```

Comment:
Is this code path even possible now? I would guess that with the change above the convergedArgType in NumericExpression must be one of Decimal, Float, or Double so the case like ``("div", Long)`` and other related int types wouldn't even be possible. So DivLong and IDivLong would never be used. I would htink only DivDecimal, DivDouble, and DivFloat need to exist?

Reply:
I threw in a quick print statement the code path does get hit still, so I think this function and the others are still necessary.

Reply:
Yeah, after taking a look I see now that the convergedArgType is based on the arguemtns and not the expected result. So Div{Long,Int,etc} are divisions based on the arguments.

However, I think that means Div{Integer-like things} needs to be updated, similar to the change you've made for DivLong. So DivLong, DIvShort, DIvUnsignedBYte, etc. all have the same issue.